### DISCUSSION TITLE
Please do not install tests

### DISCUSSION THREAD

**Discussion by yurivict** [searx/searx#2548] [Installation issue: custom]
I am getting the message:
```
pkg-static: py27-searx-0.14.0 conflicts with py27-suds-jurko-0.6 (installs files into the same place).  Problematic file: /usr/local/lib/python2.7/site-packages/tests/__init__.py
```

There is no need to install ```tests```; they only conflict with other packages that do the same.

Version 0.14.0.

**Comment by asciimoo** [comment]
Do you use the debian packaged version of searx?

**Comment by yurivict** [comment]
No, I am creating the FreeBSD port, and it can't be installing tests.

**Comment by josch** [comment]
In the Debian package we just patch searx to not install tests. In fact we don't install lots of things. Our full diff against `setup.py` reads:

```diff
diff --git a/setup.py b/setup.py
index 6466d1f..1b0ae30 100644
--- a/setup.py
+++ b/setup.py
@@ -51,23 +51,9 @@ setup(
     },
     package_data={
         'searx': [
-            'settings.yml',
-            '../README.rst',
-            '../requirements.txt',
-            '../requirements-dev.txt',
             'data/*',
             'plugins/*/*',
-            'static/*.*',
-            'static/*/*.*',
-            'static/*/*/*.*',
-            'static/*/*/*/*.*',
-            'static/*/*/*/*/*.*',
-            'templates/*/*.*',
-            'templates/*/*/*.*',
-            'tests/*',
-            'tests/*/*',
-            'tests/*/*/*',
-            'translations/*/*/*'
+            'translations/*/*/*.mo'
         ],
     },
```

**Comment by yurivict** [comment]
I patched the upcoming FreeBSD package too. This bug is meant to "upstream" this, so that distros don't all need to patch.

---

BTW, ```settings.yml```, ```templates``` are required files. You shouldn't have patched them away.

**Comment by josch** [comment]
We patch them out in Debian because:

 - `settings.yml` is a configuration file and thus belongs into `/etc/searx`. We also ship `/usr/share/doc/searx/examples/settings.yml.gz`

 - `templates` and `static` would be installed into `/usr/lib/python3/dist-packages/searx` if they were not commented out in `setup.py`. But those are static files which are not Python scripts and thus they belong to `/usr/share/searx` instead.

**Comment by yurivict** [comment]
> settings.yml is a configuration file and thus belongs into /etc/searx

Exactly: https://github.com/asciimoo/searx/issues/1211

> But those are static files which are not Python scripts and thus they belong to /usr/share/searx instead.

https://github.com/asciimoo/searx/issues/1212
